src/platform_linux/shmem.c

```diff
@@ -224,17 +226,93 @@ platform_valid_addr_in_shm(platform_heap_handle heap_handle, void *addr)
 }
 
 /*
- * platform_valid_addr_in_heap(), platform_valid_addr_in_shm()
- *
  * Validate that input address 'addr' is a valid address within shared segment
  * region.
  */
-static inline bool
-platform_valid_addr_in_heap(platform_heap_id heap_id, void *addr)
+bool
```

Comment:
In part-2, the isn't much real logic change in `shmem.c`, other than adding more tracing, testing hooks and better metrics.